lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module

```diff
@@ -90,6 +90,32 @@ function dosomething_organ_donation_post_registration() {
 
   $response = drupal_http_request($client['base_url'] . '/v2/registrations/', $options);
 
+  // Load the user so we can make updates
+  global $user;
+  $account = user_load($user->uid);
```

Comment:
`global $user` should be the full loaded object, you shouldn't need to re-load here